lrauv_ignition_plugins/scripts/make_dem_utm_tiles.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2022 Monterey Bay Aquarium Research Institute (MBARI)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Development of this module has been funded by the Monterey Bay Aquarium
+# Research Institute (MBARI) and the David and Lucile Packard Foundation
+
+import subprocess
+
+import netCDF4 as nc
+import numpy as np
+from pyproj import Proj
+
+
+def mk_tiles(map_path: str, utm_zone: int, out_path: str, tile_size: int, overlap: int):
+    """
+    Generates digital elevation map (DEM) tiles from parent DEM using Generic Mapping Tool (GMT).
+
+    To install GMT,see: https://www.generic-mapping-tools.org
+
+    Tiles are generated with size of tile_size x tile_size, and with "overlap"
+    meters shared with adjacent tiles on each edge.
+
+    Tile boundaries will be computed starting at the bottom/right corner of
+    the parent DEM (rounded to nearest 100 m).
+
+    :param map_path: file path to full-resolution DEM
+    :param utm_zone: full-res DEM UTM zone
+    :param out_path: output path for tile files
+    :param tile_size: meters, each tile will have are of tile_size x tile_size
+    :param overlap: meters, tiles will have "overlap" meters shared with adjacent tiles on each edge
+    :return: void
+    """
+    unique_size = tile_size - 2 * overlap
+
+    # Open the file and create an instance of the ncCDF4 class
+    nc_fid = nc.Dataset(map_path, 'r')
+
+    # Extract data from NetCDF file
+    x = nc_fid.variables['x'][:]
+    y = nc_fid.variables['y'][:]
+    z = nc_fid.variables['z'][:]
+
+    # Find DEM boundary limits (to nearest 100)
+    x_lim = [np.ceil(min(x) / 100) * 100, np.floor(max(x) / 100) * 100]
```

Comment:
@braanan meta: this isn't strictly to the nearest mark. Either way, I'm curious: shouldn't these intervals include all points?